### PR TITLE
cmd/dlv: add shell (non-)completions to flags taking args

### PR DESCRIPTION
**Pull request by scop**
Mostly to avoid the default filename completions to flags that don't operate on filenames. Then again, mark ones that do explicitly as such, and add more specific completions for a number of other cases, too.

### CODE REVIEW

File: cmd/dlv/cmds/commands.go [hunk 1]
```diff
@@ -138,21 +138,48 @@ func New(docCall bool) *cobra.Command {
 	}
 
 	rootCommand.PersistentFlags().StringVarP(&addr, "listen", "l", "127.0.0.1:0", "Debugging server listen address. Prefix with 'unix:' to use a unix domain socket.")
+	if err := rootCommand.RegisterFlagCompletionFunc("listen", cobra.NoFileCompletions); err != nil {
```
Comment (aarzilli) [inline cmd/dlv/cmds/commands.go:141]: add 

```
func must(err error) {
    if err != nil {
        log.Fatal(err)
    }
}
```

and wrap all of this calls with it (or we could also not check the error).

Comment (scop) [reply cmd/dlv/cmds/commands.go]: Done. Would rather check the error as the failure cases are delve side programming errors.

**Review by aarzilli — APPROVED** [review]
LGTM

**Review by derekparker — APPROVED** [review]
LGTM

**Pull request merged**